### DISCUSSION TITLE
Welcome to bitnet.cpp Discussions!

### DISCUSSION THREAD

**Discussion by shumingma** [microsoft/BitNet#16] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by surajssc1232** [comment]
what does 1-bit means

**Reply by Gabrielfernandes7** [reply]
In this context, "1-bit" refers to the precision or the number of bits used to represent each parameter or value within the model. Traditionally, AI models use higher precision (such as 16 or 32 bits per value), but in this case, BitNet uses a 1-bit representation, which is much more compact. This allows the model to be significantly more efficient in terms of memory usage and speed, while maintaining the precision of the inference process as "lossless."

**Reply by surajssc1232** [reply]
how can i use it and what is the minimum specs

**Comment by Theajoseph** [comment]
bonsoir

**Comment by grctest** [comment]
Hey, I'm a full-stack dev, highly interested in the potential of 1-bit LLM's to deliver major electricity savings and performance optimizations, I've thrown together a [BitNet Electron app](https://github.com/microsoft/BitNet/discussions/85) to run the inference executable without using a developer terminal, hoping to gain more dev experience where I can!

Thanks for creating this technology, keep up the good work!

**Reply by Gabrielfernandes7** [reply]
Good job!

**Comment by AnandKumarSinha07** [comment]
it is interesting to get connect with great developer

**Comment by Wiwuth** [comment]
> ## 👋 Welcome!
> We’re using Discussions as a place to connect with other members of our community. We hope that you:
> 
> * Ask questions you’re wondering about.
> * Share ideas.
> * Engage with other community members.
> * Welcome others and are open-minded. Remember that this is a community we
>   build together 💪.
> 
> To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by y-vectorfield** [comment]
I ask a first question.
I submitted three PRs in this repository last month regarding urgent issues.(#163, #164)
However, these have never been reviewed yet. What do you think I should do?
I would like to get feedbacks for these.

**Comment by preethi-github-29** [comment]
Hello,

I'm currently working with the BitNet repository and have successfully downloaded the config.json and model.safetensors weight files.
However, I'm finding that the Python code defining the BitNetForCausalLM model class (typically found in a file like modeling_bitnet.py or similar) appears to be missing from the repository.
Without this core model definition, it's not possible to properly instantiate the BitNet architecture in PyTorch and load the model.safetensors weights. This prevents further exploration and use of the pre-trained model.
Could you please provide the necessary Python code (the modeling_bitnet.py file or equivalent) that defines the BitNetForCausalLM class and its components, so that the model.safetensors weights can be loaded correctly?
